packages/react/src/components/Carousel/Carousel.tsx

```diff
@@ -273,19 +299,49 @@ class Carousel extends AutoControlledComponent<WithAsProp<CarouselProps>, Carous
             items.map((item, index) => {
               const itemRef = React.createRef<HTMLElement>()
               this.itemRefs.push(itemRef)
+              const active = activeIndex === index
+              let slideToNext = prevActiveIndex < activeIndex
+
+              const initialMounting = prevActiveIndex === -1
+
+              if (circular && prevActiveIndex === items.length - 1 && activeIndex === 0) {
+                slideToNext = true
+              } else if (circular && prevActiveIndex === 0 && activeIndex === items.length - 1) {
+                slideToNext = false
+              }
+
               return (
-                <Ref key={item['key'] || index} innerRef={itemRef}>
-                  {CarouselItem.create(item, {
-                    defaultProps: () => ({
-                      active: activeIndex === index,
-                      id: itemIds[index],
-                      navigation: !!this.props.navigation,
-                      ...(getItemPositionText && {
-                        itemPositionText: getItemPositionText(index, items.length),
+                <Animation
```

Comment:
Just wondering why this thing is not a part of `CarouselItem`?

Reply:
Some of the things here are specific to the Carousel, like whether we are sliding to next/prev, whether it is initialMounting etc. We would need to add more props to the CarouselItem, for handling these. I'd rather keep this here and keep the CarouselItem be responsible only for the things it should know about

Reply:
Because if we will decide to have there children API there will be animations only for shorthands...

Reply:
Still not sure that it is a good way of handling it, as I would rather wrap anything with Animations, than adding Animation in any component, just because it may be animated somewhere... We can iterate when we will add children and consider what is the best option that would work in both cases.